identity/cache.go

```diff
@@ -15,23 +16,24 @@ type IdentityCache struct {
 	File string
 }
 
-func NewIdentityCache(dir string, jsonFile string) *IdentityCache {
+func NewIdentityCache(dir string, jsonFile string) IdentityCacheInterface {
 	return &IdentityCache{
 		File: filepath.Join(dir, jsonFile),
 	}
 }
 
 func (ic *IdentityCache) GetIdentity() (identity Identity, err error) {
-	if ic.cacheExists() {
-		cache, err := ic.readCache()
-		if err != nil {
-			return identity, err
-		}
+	if !ic.cacheExists() {
+		err = errors.New("cache file does not exist")
```

Comment:
It's not an error if file is missing. It's same as "identity non found in cache"

Reply:
1. This method is named `GetIdentity` and it seems a common sense to get an identity or an error.

2. Looking in `identity_handler.go`:
```go
	identity, err = ih.cache.GetIdentity()
	if err != nil || !ih.manager.HasIdentity(identity.Address) {
		return identity, errors.New("identity not found in cache")
	}
```
If cache file does not exist, this results in "identity not found in cache" error either way - with or without this change.
What this change is about is not only making method more predictable, but also avoiding calling
`ih.manager.HasIdentity("")` for empty identity, which is returned when cache does not exist. If we know that cache file does not exist, we should not check for an empty identity in identity manager.